lib/elixir/lib/kernel.ex

```diff
@@ -2292,8 +2292,18 @@ defmodule Kernel do
   In order to compare more than two clauses, the `cond/1` macro has to be used.
   """
   defmacro if(condition, clauses) do
-    do_clause = Keyword.get(clauses, :do, nil)
-    else_clause = Keyword.get(clauses, :else, nil)
+    {do_clause, clauses} = Keyword.pop_first(clauses, :do)
```

Comment:
As an aside - checking for the presence of `do` means that the implementation of `unless` will have to be modified as it called `if` with the keys swapped.